bindings/jaxrs/src/main/java/org/commonjava/indy/core/bind/jaxrs/admin/StoreAdminHandler.java

```diff
@@ -477,4 +490,85 @@ public Response getRemoteByUrl( final @PathParam( "packageType" ) String package
         return response;
     }
 
+
+    @ApiOperation( "Revalidation of Artifacts Stored on demand" )
+    @ApiResponses( { @ApiResponse( code = 200, response = ArtifactStore.class, message = "Revalidation for Remote Repositories was successfull" ),
+        @ApiResponse( code = 404, message = "Revalidation is not successfull" ), } )
+    @Path( "/revalidate" )
+    @POST
+    public Response revalidateArtifactStores(
+        @PathParam("package") String packageType,
+        @PathParam( "type" ) String type) {
+
+        ArtifactStoreValidateData result = null;
+        Map<String,String> results = new HashMap<>();
+        Response response;
+
+        try {
+            StoreType storeType =  StoreType.get(type);
+
+            List<ArtifactStore> allArtifactStores = adminController.getAllOfType(packageType, storeType);
+
+            for(ArtifactStore artifactStore: allArtifactStores) {
+                result = storeValidator.validate(artifactStore);
+                results.put(artifactStore.getKey().toString(), result.getErrors().toString());
+
+            }
+            response = formatOkResponseWithJsonEntity(results, objectMapper);
+
+        } catch (IndyDataException ide) {
+            logger.warn("=> [IndyDataException] exception message: " + ide.getMessage());
+            response = formatResponse(ide);
+
+        } catch (MalformedURLException mue) {
+            logger.warn("=> [MalformedURLException] Invalid URL exception message: " + mue.getMessage());
+            response = formatResponse(mue);
+
+        } catch (IndyWorkflowException iwe) {
+            logger.warn("=> [IndyWorkflowException] exception message: " + iwe.getMessage());
+            response = formatResponse(iwe);
+
+        }
+        return response;
+    }
+
+    @ApiOperation( "Revalidation of Artifact Stored on demand based on package, type and name" )
+    @ApiResponses( { @ApiResponse( code = 200, response = ArtifactStore.class, message = "Revalidation for Remote Repository was successfull" ),
+        @ApiResponse( code = 404, message = "Revalidation is not successfull" ), } )
+    @Path( "/{name}/revalidate" )
```

Comment:
Again, you'll need to change the @PathParam to `packageType` to match what's in the class-level annotation.